scripts/memory/redismem.py

```diff
@@ -0,0 +1,141 @@
+        try:
+            self.redis.ft(f"{cfg.memory_index}").create_index(
+                fields=SCHEMA,
+                definition=IndexDefinition(
+                    prefix=[f"{cfg.memory_index}:"],
+                    index_type=IndexType.HASH
+                    )
+                )
+        except Exception as e:
+            print("Error creating Redis search index: ", e)
+        existing_vec_num = self.redis.get(f'{cfg.memory_index}-vec_num')
+        self.vec_num = int(existing_vec_num.decode('utf-8')) if\
+            existing_vec_num else 0
+
+    def add(self, data: str) -> str:
+        """
+        Adds a data point to the memory.
+
+        Args:
+            data: The data to add.
+
+        Returns: Message indicating that the data has been added.
+        """
+        vector = get_ada_embedding(data)
+        vector = np.array(vector).astype(np.float32).tobytes()
+        data_dict = {
+            b"data": data,
+            "embedding": vector
+        }
+        pipe = self.redis.pipeline()
+        pipe.hset(f"{self.cfg.memory_index}:{self.vec_num}", mapping=data_dict)
+        _text = f"Inserting data into memory at index: {self.vec_num}:\n"\
+            f"data: {data}"
+        self.vec_num += 1
+        pipe.set(f'{self.cfg.memory_index}-vec_num', self.vec_num)
+        pipe.execute()
+        return _text
+
+    def get(self, data: str) -> Optional[List[Any]]:
+        """
+        Gets the data from the memory that is most relevant to the given data.
+
+        Args:
+            data: The data to compare to.
+
+        Returns: The most relevant data.
+        """
+        return self.get_relevant(data, 1)
+
+    def clear(self) -> str:
+        """
+        Clears the redis server.
+
+        Returns: A message indicating that the memory has been cleared.
+        """
+        self.redis.flushall()
+        return "Obliviated"
+
+    def get_relevant(
+        self,
+        data: str,
+        num_relevant: int = 5
+    ) -> Optional[List[Any]]:
+        """
+        Returns all the data in the memory that is relevant to the given data.
+        Args:
+            data: The data to compare to.
+            num_relevant: The number of relevant data to return.
+
+        Returns: A list of the most relevant data.
+        """
+        query_embedding = get_ada_embedding(data)
+        base_query = f"*=>[KNN {num_relevant} @embedding $vector AS vector_score]"
+        query = Query(base_query).return_fields(
+            "data",
+            "vector_score"
+        ).sort_by("vector_score").dialect(2)
+        query_vector = np.array(query_embedding).astype(np.float32).tobytes()
+
+        try:
+            results = self.redis.ft(f"{self.cfg.memory_index}").search(
+                query, query_params={"vector": query_vector}
+            )
+        except Exception as e:
+            print("Error calling Redis search: ", e)
+            return None
+        return list(results.docs)
```

Comment:
```suggestion
        return [result.data for result in results.docs]
```

Reply:
This will make the output more consistent with the `get_relevant` method in the `scripts/memory/local.py` file and improve the AI's readability of the results.